lib/deaf_grandma.rb

```diff
@@ -14,14 +14,29 @@ def run!
     loop do
       user_input = get_user_input
       p speak(user_input)
+      exit if @bye_counter == 3
     end
   end
 
 
   def speak(input)
+    if input.match(/\p{Lower}/)
+      @bye_counter = 0
+      soft
+    else
+      if input == "BYE"
+        @bye_counter += 1
 
-    #Implement your code here <<<<<<<<<
-
+        if @bye_counter == 3
+          see_you_later
+        else
+          yell
+        end
+      else
+        @bye_counter = 0
+        yell
+      end
+    end
```

Comment:
This is nested three levels deep.  It's actually not a terrible use-case for deep nesting, but generally that's something to avoid.